spec/controller/application_controller_spec.rb

```diff
@@ -34,43 +34,28 @@ def index
       request.env["warden"] = instance_double(Warden::Proxy)
     end
 
-    context "when Signon is enabled" do
-      before do
-        # Mock GDS SSO
-        allow(warden_spy).to receive(:authenticate!).and_return(true)
-        allow(controller).to receive(:current_user).and_return(user)
-
-        allow(Settings).to receive(:auth_provider).and_return("gds_sso")
-
-        get :index
-      end
-
-      it "uses GOV.UK Signon" do
-        expect(warden_spy).to have_received(:authenticate!).with(:gds_sso)
-      end
-
-      it "sets @current_user" do
-        expect(assigns[:current_user]).to eq user
-      end
-    end
-
-    context "when basic auth is enabled" do
-      before do
-        # Mock Warden
-        allow(warden_spy).to receive(:authenticate!).and_return(true)
-        allow(controller).to receive(:current_user).and_return(user)
-
-        allow(Settings).to receive(:auth_provider).and_return("basic_auth")
-
-        get :index
-      end
-
-      it "uses HTTP Basic Authentication" do
-        expect(warden_spy).to have_received(:authenticate!).with(:basic_auth)
-      end
-
-      it "sets @current_user" do
-        expect(assigns[:current_user]).to eq user
+    %w[
+      auth0
+      basic_auth
+      gds_sso
+    ].each do |provider|
```

Comment:
I always feel like being able to create parameterised tests is an indicator that the code under test has been structured nicely 👏